pennylane_lightning/src/simulator/StateVector.hpp

```diff
@@ -106,8 +138,10 @@ template <class fp_t = double> class StateVector {
      * @param arr Pointer to the complex data array.
      * @param length Number of elements in complex data array. Must be
      * power-of-2 (qubits only).
+     * @param cache_size Cache size to store most calculated indices throughout
+     * operations.
      */
-    StateVector(CFP_t *arr, size_t length)
+    StateVector(CFP_t *arr, size_t length, size_t cache_size = 10)
```

Comment:
```suggestion
    StateVector(CFP_t *arr, size_t length, size_t cache_size = 0)
```

We should default to the cache being disabled. This also means we require to be no cache, rather than infinite.